plugin-api/src/main/java/net/clgd/ccemux/emulation/EmuConfig.java

```diff
@@ -0,0 +1,55 @@
+package net.clgd.ccemux.emulation;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.google.gson.Gson;
+import net.clgd.ccemux.config.Config;
+import net.clgd.ccemux.config.JsonAdapter;
+import net.clgd.ccemux.config.ConfigProperty;
+
+/**
+ * The root config used by CCEmuX
+ *
+ */
+public abstract class EmuConfig extends Config {
```

Comment:
There's a bit of me which doesn't like exposing this as a concrete class. Aside from the terminal information, we don't really need to expose any of the fields.

Reply:
The alternative would be writing getters, which would make it even more verbose to use - `config.getTermScale().get()`. It's not the end of the world, but it's annoying.

Reply:
I guess you could just have `config.termScale().get()` which is slightly shorter. Dammn Java and it's lack of properties :/. I guess I don't really want to include all the "meta" information such as comments and what not.